frontend/ui/widget/htmlboxwidget.lua

```diff
@@ -0,0 +1,170 @@
+--[[--
+HTML widget (without scroll bars).
+--]]
+
+local DrawContext = require("ffi/drawcontext")
+local Geom = require("ui/geometry")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local Mupdf = require("ffi/mupdf")
+local TimeVal = require("ui/timeval")
+
+local HtmlBoxWidget = InputContainer:new{
+    bb = nil,
+    dimen = nil,
+    document = nil,
+    page_count = 0,
+    page_number = 1,
+    hold_start_pos = nil,
+    hold_start_tv = nil,
+}
+
+function HtmlBoxWidget:setContent(body, css, default_font_size)
+    -- fz_set_user_css is tied to the context instead of the document so to easily support multiple
+    -- HTML dictionaries with different CSS, we embed the stylesheet into the HTML instead of using
+    -- that function.
+    local html = "<html>"
+    if css then
+        html = html .. "<head><style>" .. css .. "</style></head>"
```

Comment:
I'm a bit tired and need to feed the cats but if there's a way to rewrite this section with string.format that could be an idea. (It's faster in LuaJIT.) Not a biggie of course.

Reply:
The condition now only sets the `<head>`, it made the code clearer too.